micro_pumas_v1.F90

```diff
@@ -2038,32 +2038,29 @@ subroutine micro_pumas_tend ( &
   !$acc end parallel
 
   if (do_cldice) then
+     if (.not. use_hetfrz_classnuc) then
+        ! heterogeneous freezing of cloud water
+        !----------------------------------------------
+        call immersion_freezing(microp_uniform, t, pgam, lamc, qcic, ncic, relvar, mnuccc, nnuccc, mgncol*nlev)
 
```

Comment:
So now when user_hetfrz_classnuc is false, then we call immersion_freezing. But since user_hetfrz_classnuc is true for our workhorse configuration, then it's no longer invoking this subroutine immersion_freezing. Does that mean this immersion_freezing subroutine is the Bigg's scheme, that we are no longer using as a result of this PR + the CAM PR where we limit the dust available to classnuc theory?

Reply:
Yes, the immersion_freezing subroutine is labeled in the comment above it "Bigg, 1953", and it will no longer be called when use_hetfrz_classnuc is true (which is default for cam6 and cam_dev physics). And Francis is handling the CAM PR so I can't be 100% positive about what's in there, but I expect it will include the dust changes.

Reply:
ok. this revert actually makes things less wonky in the code. His temporary fix to use classnuc except for immersion freezing is what was tricky. Now that he's putting immersion_freezing back under the .not. user_hetfrz_classnuc, it's easier to see what's happening.

I do not see any text with "Bigg" in it when I grep'd the pumas directory. Is it right under my nose? If it's missing could you please add that reference above the immersion_freezing call? Thx!

Reply:
Hmm, not sure what your grep was, but it's line 1575 in micro_pumas_utils.F90 : https://github.com/ESCOMP/PUMAS/blob/katetc/ice_nuc_update/micro_pumas_utils.F90#L1575

I'll add a comment around the call in micro_pumas_v1.F90 as well.